pkg/virt-launcher/virtwrap/cmd-server/server.go

```diff
@@ -310,7 +310,14 @@ func (l *Launcher) GetDomain(ctx context.Context, request *cmdv1.EmptyRequest) (
 	}
 
 	if len(list) > 0 {
-		if domain, err := json.Marshal(list[0]); err != nil {
+		domainObj := list[0]
+		if osInfo := l.domainManager.GetGuestOSInfo(); osInfo != nil {
+			domainObj.Status.OSInfo = *osInfo
+		}
+		if interfaces := l.domainManager.InterfacesStatus(domainObj.Spec.Devices.Interfaces); interfaces != nil {
+			domainObj.Status.Interfaces = interfaces
+		}
```

Comment:
If we don't share this logic with `pkg/virt-launcher/notify-client/client.go`'s `eventCallback` function, which sends the domain events, then we're going to have the same problem the next time we try to inject status info into the domain object before passign it back to virt-handler

This PR technically works. I don't think it a sustainable solution to the problem though.